src/project.ts

```diff
@@ -97,6 +97,7 @@ export interface Browser {
   revision: string | null;
   peer: string | null;
   path: string | null;
+  hash: string | null;
```

Comment:
Isn't it better to store `line: number | null`?

Reply:
I didn't want to commit myself to `line: number | null` since we eventually could use the location hash for pointing to a commit diff (similar to github using e.g. https://github.com/radicle-dev/radicle-client-services/pull/122/files#diff-0b5c61cb7e37875016664c483ddc7eb6947e7019b710684c9ffc956a79a80491) or other places

Reply:
When that happens we can turn the type into an enum or something, eg. `{ line: number } | { diff: ... } | null`. But you're currently mixing domains: revision, peer and path are all meaningful data about a code resource, and then you have `hash` which has to do with how web browsers work. These fundamentally don't belong in the same type.

Reply:
Ok got it, yeah makes sense 👍